core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/federation/RepositoryFederatedService.java

```diff
@@ -419,13 +503,28 @@ public void shutdown() throws QueryEvaluationException {
 		}
 	}
 
-	protected RepositoryConnection getConnection() throws RepositoryException {
-		// use a cache connection if possible
-		// (TODO add mechanism to unset/close connection)
-		if (conn == null) {
-			conn = rep.getConnection();
+	/**
+	 * Return a fresh {@link RepositoryConnection} from the configured repository.
+	 * 
+	 * @return
+	 * @throws RepositoryException
+	 */
+	protected RepositoryConnection freshConnection() throws RepositoryException {
```

Comment:
Same here: does this need to be protected? Or can it be made private?

Reply:
done, made to private